src/bots/pins/index.js

```diff
@@ -1,4 +1,26 @@
 import client from '../../client';
-import handler from './handler';
+import { commandHandler } from '../../utils';
+import listPins from './subCommands/listPins';
+import pinByReact from './subCommands/pinByReact';
+import { getMemberFromUser, isContributor, isMod } from '../../utils/perms';
 
-client.on('pins', handler);
+const subCommands = { list: listPins };
+
+client.on('pins', () => commandHandler(subCommands, {}));
+
+client.on('messageReactionAdd', async (reaction) => {
+  reaction.users.cache.each(async (user) => {
+    const member = await getMemberFromUser(user);
+    if (isContributor(member) || isMod(member)) {
```

Comment:
What would happen under the following case scenario?
- Person A (contributor) pins the message
- Person B (non-contributor) pins the message then unpins
- Will this unpin the message since one person is a contributor?

Reply:
oh i just realized i didnt add this permissioning stuff to the remove reaction event. bullet point 3 will be a no once i add that in

Reply:
Would make this moot then
https://github.com/cscareerhub/megabot/pull/70/files#diff-2cc7a9d084399a1d0b427b8c10e0b49cf909ec33edb4aff7a9b0c90894cb1f52R10

Reply:
not seeing what that link is referring to?

Reply:
don't see any line highlighted or anything

Reply:
Line 25 in pinByReaction.js